src/Adapter/Product/Image/Repository/ProductImageRepository.php

```diff
@@ -364,4 +392,76 @@ public function delete(Image $image): void
     {
         $this->deleteObjectModel($image, CannotDeleteProductImageException::class);
     }
+
+    /**
+     * @param ProductId $productId
+     * @param LanguageId $languageId
+     *
+     * @return ProductPreview
+     *
+     * @throws CoreException
+     */
+    public function getProductCoverUrl(ProductId $productId, LanguageId $languageId): ProductPreview
+    {
+        $product = $this->productRepository->get($productId);
+        $name = $product->name[$languageId->getValue()] ?? ($product->name[$this->defaultLanguageId] ?? reset($product->name));
+        $imageId = $this->getDefaultImageId($productId);
+        $imagePath = $imageId ?
+            $this->productImagePathFactory->getPath($imageId, ProductImagePathFactory::DEFAULT_IMAGE_FORMAT) :
+            $this->productImagePathFactory->getNoImagePath(ProductImagePathFactory::IMAGE_TYPE_SMALL_DEFAULT)
+        ;
+
+        return new ProductPreview(
+            $productId->getValue(),
+            $name,
+            $imagePath
+        );
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param LanguageId $languageId
+     * @param ?CombinationId $combinationId
+     *
+     * @return ProductPreview
+     *
+     * @throws CoreException
+     */
+    public function getCombinationCoverUrl(ProductId $productId, LanguageId $languageId, ?CombinationId $combinationId = null): ProductPreview
```

Comment:
This method is targetting a combination so the `CombinationId` should be mandatory, it's up to the code that is using this method to make sure it is handling a combination or a product